scipy/stats/_relative_risk.py

```diff
@@ -0,0 +1,266 @@
+
+    Methods
+    -------
+    confidence_interval(confidence_level=0.95)
+        Compute the confidence interval of the relative risk for a given
+        confidence level.
+
+    """
+    def __init__(self, exposed_cases, exposed_total,
+                 control_cases, control_total,
+                 relative_risk):
+        self.relative_risk = relative_risk
+        self.exposed_cases = exposed_cases
+        self.exposed_total = exposed_total
+        self.control_cases = control_cases
+        self.control_total = control_total
+
+    def __repr__(self):
+        return f"RelativeRiskResult(relative_risk={self.relative_risk})"
+
+    def confidence_interval(self, confidence_level=0.95):
+        """
+        Compute the confidence interval for the relative risk.
+
+        The confidence interval is computed using the Katz method
+        ([1]_, [2]_).
+
+        Parameters
+        ----------
+        confidence_level : float, optional
+            The confidence level to use for the confidence interval.
+            Default is 0.95.
+
+        Returns
+        -------
+        ci : ConfidenceInterval instance
+            The return value is an object with attributes ``low`` and
+            ``high`` that hold the confidence interval.
+
+        References
+        ----------
+        .. [1] Hardeo Sahai and Anwer Khurshid, Statistics in Epidemiology,
+               CRC Press LLC, Boca Raton, FL, USA (1996).
+        .. [2] D. Katz, J. Baptista, S. P. Azen and M. C. Pike, "Obtaining
+               confidence intervals for the risk ratio in cohort studies",
+               Biometrics, 34, 469-474 (1978).
+
+        Examples
+        --------
+        >>> from scipy.stats.contingency import relative_risk
+        >>> result = relative_risk(exposed_cases=10, exposed_total=75,
+        ...                        control_cases=12, control_total=225)
+        >>> result.relative_risk
+        2.5
+        >>> result.confidence_interval()
+        ConfidenceInterval(low=1.1261564003469628, high=5.549850800541033)
+        """
+        if not 0 <= confidence_level <= 1:
+            raise ValueError('confidence_level must be in the interval '
+                             '[0, 1].')
+
+        # Handle edge cases where either exposed_cases or control_cases
+        # is zero.  We follow the convention of the R function riskratio
+        # from the epitools library.
+        if self.exposed_cases == 0 and self.control_cases == 0:
+            # relative risk is nan.
+            return ConfidenceInterval(low=np.nan, high=np.nan)
+        elif self.exposed_cases == 0:
+            # relative risk is 0.
+            return ConfidenceInterval(low=0.0, high=np.nan)
+        elif self.control_cases == 0:
+            # relative risk is inf
+            return ConfidenceInterval(low=np.nan, high=np.inf)
+
+        alpha = 1 - confidence_level
+        z = ndtri(1 - alpha/2)
```

Comment:
I assume that the convention Katz follows for defining confidence limits is different, as naiive implementation of Katz would be `(1 - alpha)`. Might be worth noting. I assume this is because Katz seems to think of the upper and lower confident limits separately, each with confidence (1-alpha), whereas we think of a confidence interval with confidence (1-alpha)?

Reply:
This gives the interval around the mean of a normal distribution, i.e. it is a two-sided calculation.  Each tail gets half of alpha, so the confidence interval covers `confdence_level` and is centered on the mean.  See, e.g., step 2 with a known σ in [Confidence interval / Basic steps](https://en.wikipedia.org/wiki/Confidence_interval#Basic_steps).